outport/process/executionOrder/transactionsExecutionOrder.go

```diff
@@ -0,0 +1,258 @@
+	"github.com/ElrondNetwork/elrond-go-core/data/outport"
+	"github.com/ElrondNetwork/elrond-go-core/data/smartContractResult"
+	"github.com/ElrondNetwork/elrond-go-core/hashing"
+	logger "github.com/ElrondNetwork/elrond-go-logger"
+	"github.com/ElrondNetwork/elrond-go/process"
+)
+
+var log = logger.GetOrCreate("outport/process/executionOrder")
+
+type sorter struct {
+	hasher hashing.Hasher
+}
+
+func NewSorter(hasher hashing.Hasher) (*sorter, error) {
+	if check.IfNil(hasher) {
+		return nil, process.ErrNilHasher
+	}
+
+	return &sorter{
+		hasher: hasher,
+	}, nil
+}
+
+func (s *sorter) PutExecutionOrderInTransactionPool(
+	pool *outport.Pool,
+	header data.HeaderHandler,
+	body data.BodyHandler,
+) error {
+	blockBody, ok := body.(*block.Body)
+	if !ok {
+		log.Warn("s.PutExecutionOrderInTransactionPool cannot cast BodyHandler to *Body")
+		return nil
+	}
+
+	// already sorted
+	transactionsToMe := extractNormalTransactionAndScrsToMe(pool, blockBody, header, true)
+
+	// scheduled to me, already sorted
+	scheduledTransactionsToMe := extractNormalTransactionAndScrsToMe(pool, blockBody, header, false)
+
+	// need to be sorted
+	transactionsFromMe := extractNormalTransactionsAndInvalidFromMe(pool, blockBody, header, true)
+	txsSort.SortTransactionsBySenderAndNonceWithFrontRunningProtectionExtendedTransactions(transactionsFromMe, s.hasher, header.GetPrevRandSeed())
+
+	rewardsTxs := getRewardsTxsFromMe(pool, blockBody, header)
+
+	// scheduled from me, need to be sorted
+	scheduledTransactionsFromMe := extractNormalTransactionsAndInvalidFromMe(pool, blockBody, header, false)
+	txsSort.SortTransactionsBySenderAndNonceWithFrontRunningProtectionExtendedTransactions(scheduledTransactionsFromMe, s.hasher, header.GetPrevRandSeed())
+
+	allTransaction := append(transactionsToMe, scheduledTransactionsToMe...)
+	allTransaction = append(allTransaction, transactionsFromMe...)
+	allTransaction = append(allTransaction, rewardsTxs...)
+	allTransaction = append(allTransaction, scheduledTransactionsFromMe...)
+
+	for idx, tx := range allTransaction {
+		tx.SetExecutionOrder(idx)
+	}
+
+	setOrderSmartContractResults(pool)
+
+	prinPool(pool)
+	return nil
+}
+
+func setOrderSmartContractResults(pool *outport.Pool) {
+	for _, scrHandler := range pool.Scrs {
+		scr, ok := scrHandler.GetTxHandler().(*smartContractResult.SmartContractResult)
+		if !ok {
+			continue
+		}
+
+		tx, found := pool.Txs[string(scr.OriginalTxHash)]
+		if !found {
+			continue
+		}
+
+		scrHandler.SetExecutionOrder(tx.GetExecutionOrder())
+	}
+}
+
+func extractNormalTransactionsAndInvalidFromMe(pool *outport.Pool, blockBody *block.Body, header data.HeaderHandler, ignoreScheduled bool) []data.TransactionHandlerWithGasUsedAndFee {
+	normalTxsHashes := make([][]byte, 0)
+	invalidTxsHashes := make([][]byte, 0)
+	for mbIndex, mb := range blockBody.MiniBlocks {
+		if isMBScheduled(header, mbIndex) == ignoreScheduled || shouldIgnoreProcessedMBScheduled(header, mbIndex) {
```

Comment:
please refactor this, either create separate functions or something. Extremely hard to understand
Please remove the bool parameter from here

Reply:
done.